pages/astro.yml

```diff
@@ -0,0 +1,52 @@
+# Sample workflow for building and deploying an Astro site to GitHub Pages
+#
+# To get started with Astro see: https://docs.astro.build/en/getting-started/
+#
+
+name: Deploy Astro site to Pages
+
+on:
+  push:
+    branches: [main]
+    # paths:
+    # only needed when project in subfolder
+    # - "subfolder"
+    # - ".github/workflows/astro.yml"
+  workflow_dispatch:
+
+permissions:
```

Comment:
permissions at workflow level can be `contents:read` and at job level add these permissions